### ISSUE TITLE
How to debug custom templates

### ISSUE THREAD

**Issue by weedkiller** [mono/t4#35]
Hello I am using this for custom views in asp mvc, but unable to debug in VS 2017 professional

**Comment by mhutch** [comment]
How are you running the template engine?

**Comment by weedkiller** [comment]
I put  debugger name space and  inside the template code and ran it. like here https://docs.microsoft.com/en-us/visualstudio/modeling/walkthrough-debugging-a-text-template-that-accesses-a-model?view=vs-2017

I am trying to scaffold controllers

**Comment by devlead** [comment]
T4 tool doesn't have the full VS hosting model, but keeping symbols file can be done by setting debug to `true` in the template directive i.e.
`<#@ template debug="true" language="C#" #>`

With the symbols in place you could attach process and debug the template, or use system diagnostic to break/attach debugger etc.

Example
```csharp
<#@ template debug="true" language="C#" #>
<#
    System.Diagnostics.Debugger.Break();
#>
<#="Hello" #>
<#
for(var i = 0;i < 10;i++)
{
    #><#=i #><#
}
#>
```

There's currently an issue with the global tool and debug which I've proposed a fix for with pr  #40.
